File: leetcode/python/1637_widest_vertical_area_between_two_points_containing_no_points.py
class Solution:
    # [[3,1],[9,0],[1,0],[1,4],[5,3],[8,8]]
    # [1,0], [1,4], [3,1], [5,3], [8,8], [9,0]
    def maxWidthOfVerticalArea(self, points: list[list[int]]) -> int:
        points.sort(key=lambda x: x[0])
        res = 0
        for i in range(1, len(points)):
            res = max(res, points[i][0] - points[i-1][0])
        return res
        

File: leetcode/rust/src/1637_widest_vertical_area_between_two_points_containing_no_points.rs
impl Solution {
    pub fn max_width_of_vertical_area(points: Vec<Vec<i32>>) -> i32 {
        let mut points = points;
        points.sort_unstable_by_key(|v| v[0]);
        let mut res = 0;
        for i in 1..points.len() {
            res = res.max(points[i][0] - points[i-1][0]);
        }
        res
    }
}